controller/konnect/ops/ops_controlplane.go

```diff
@@ -113,7 +131,104 @@ func updateControlPlane(
 	}
 
 	cp.Status.SetKonnectID(*resp.ControlPlane.ID)
+
+	if err := setGroupMembers(ctx, cl, cp, sdkGroups); err != nil {
+		SetKonnectEntityProgrammedConditionFalse(cp, conditions.KonnectGatewayControlPlaneProgrammedReasonFailedToSetControlPlaneGroupMembers, err.Error())
+		return err
+	}
+
 	SetKonnectEntityProgrammedCondition(cp)
 
 	return nil
 }
+
+func setGroupMembers(
```

Comment:
Have you considered using a library to handle concurrency instead of implementing the producer/consumer here? I think that [iter.MapErr](https://pkg.go.dev/github.com/sourcegraph/conc@v0.3.0/iter#MapErr) from `sourcegraph/conc` that we used for quite a long time in [KIC](https://github.com/Kong/kubernetes-ingress-controller/blob/f8039f19b0f2e0bd3d35ba9a0a688fe208d8f6f8/internal/dataplane/kong_client.go#L698) might simplify this code significantly.

Reply:
It seems that we do not import `sourcegraph.conc` just yet in KGO so I'd suggest to not pull in that dependency just for the use case above. WDYT?

Reply:
I think it's a valid use case to at least consider pulling it in - IMO if we can use a building block for concurrency stuff, it's cheaper and safer to use it than worry about maintaining our scattered implementations (where the chances of making a mistake rise).

Reply:
Makes sense 👍 801651d8a247901ab189628aab7fa8b6ef3e2cde should handle that.